app/src/main/java/com/smlnskgmail/jaman/hashcheckerlite/MainActivity.java

```diff
@@ -163,4 +194,49 @@ public void onBackPressed() {
         }
     }
 
+    private void checkForUpdateAvailability() {
+        appUpdateManager = AppUpdateManagerFactory.create(this);
+        Task<AppUpdateInfo> appUpdateInfoTask = appUpdateManager.getAppUpdateInfo();
+        appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE
+                    && appUpdateInfo.isUpdateTypeAllowed(AppUpdateType.FLEXIBLE)) {
+                // Request the update.
+                requestUpdate(appUpdateInfo, AppUpdateType.FLEXIBLE, this, MY_REQUEST_CODE);
+            }
+        });
+    }
+
+    private void requestUpdate(AppUpdateInfo appUpdateInfo, int updateType, Context context, int requestCode) {
+
+        // monitor the state of an update
+        InstallStateUpdatedListener listener = state -> {
+            if (state.installStatus() == InstallStatus.DOWNLOADED) {
+                // After the update is downloaded, show a snackbar
+                // and request user confirmation to restart the app.
+                popupSnackbarForCompleteUpdate();
+            }
+        };
+
+        try {
+            appUpdateManager.startUpdateFlowForResult(
+                    appUpdateInfo,
+                    updateType,
+                    (Activity) context,
+                    requestCode);
+        } catch (IntentSender.SendIntentException e) {
+            e.printStackTrace();
+        }
+    }
+
+    // Displays the snackbar notification and call to action.
+    private void popupSnackbarForCompleteUpdate() {
+        Snackbar snackbar =
+                Snackbar.make(
+                        findViewById(android.R.id.content).getRootView(),
+                        "An update has just been downloaded.",
```

Comment:
You need to move all strings in UI to `strings.xml` file in resources. I'm translating this sentences to other languages.